lib/rules/v-if-else-key.js

```diff
@@ -0,0 +1,299 @@
+ * @typedef {Object} ConditionalFamily
+ * @property {VElement} if - The node associated with the 'v-if' directive.
+ * @property {VElement[]} elseIf - An array of nodes associated with 'v-else-if' directives.
+ * @property {VElement | null} else - The node associated with the 'v-else' directive, or null if there isn't one.
+ */
+
+/**
+ * Map to store conditionally rendered components and their respective conditional directives.
+ *
+ * @type {Map<VElement, ConditionalFamily>}
+ */
+const conditionalFamilies = new Map()
+
+// =============================================================================
+// Rule Helpers
+// =============================================================================
+
+/**
+ * Checks for the presence of a 'key' attribute in the given node. If the 'key' attribute is missing
+ * and the node is part of a conditional family a report is generated.
+ * The fix proposed adds a unique key based on the component's name and count,
+ * following the format '${kebabCase(componentName)}-${componentCount}', e.g., 'some-component-2'.
+ *
+ * @param {VElement} node - The Vue component node to check for a 'key' attribute.
+ * @param {RuleContext} context - The rule's context object, used for reporting.
+ * @param {string} componentName - Name of the component.
+ * @param {string} uniqueKey - A unique key for the repeated component, used for the fix.
+ */
+const checkForKey = (node, context, componentName, uniqueKey) => {
+  if (node.parent && node.parent.type === 'VElement') {
+    const conditionalFamily = conditionalFamilies.get(node.parent)
+
+    if (
+      conditionalFamily &&
+      (utils.hasDirective(node, 'bind', 'key') ||
+        utils.hasAttribute(node, 'key') ||
+        !hasConditionalDirective(node) ||
+        !isConditionalFamilyComplete(conditionalFamily))
+    ) {
+      return
+    }
+
+    context.report({
+      node: node.startTag,
+      loc: node.startTag.loc,
+      messageId: 'requireKey',
+      data: {
+        componentName
+      },
+      fix(fixer) {
+        const afterComponentNamePosition =
+          node.startTag.range[0] + componentName.length + 1
+        return fixer.insertTextBeforeRange(
+          [afterComponentNamePosition, afterComponentNamePosition],
+          ` key='${uniqueKey}'`
+        )
+      }
+    })
+  }
+}
+
+/**
+ * Checks for the presence of conditional directives in the given node.
+ *
+ * @param {VElement} node - The node to check for conditional directives.
+ * @returns {boolean} Returns true if a conditional directive is found in the node or its parents,
+ *   false otherwise.
+ */
+const hasConditionalDirective = (node) =>
+  utils.hasDirective(node, 'if') ||
+  utils.hasDirective(node, 'else-if') ||
+  utils.hasDirective(node, 'else')
+
+/**
+ * Checks whether a conditional family is complete.
+ * A conditional family is considered complete if:
+ * 1. An 'if' condition is present.
+ * 2. All elements of the family (if, else, else-if) share the same rawName.
```

Comment:
Why should the rule not trigger for (at least) the first two lines in this case?

```html
<component-a v-if="foo" />
<component-a v-else-if="bar" />
<component-b v-else />
```

Or (at least) the first and the last line in this case:

```html
<component-a v-if="foo" />
<component-b v-else-if="bar" />
<component-a v-else />
```

Reply:
Hi Flo, sorry for the delayed response, things got a little hectic. Anyhow, the rule detects the first case, I'm not sure about the second one, but I'll examine this in detail as soon as I get a chance. Thanks again for making time to review this.

Reply:
Well, I am too curious and stubborn to leave this sort of question unanswered - I could not focus on anything else with this on my mind so I went back and studied the code, and you were correct to point this out, the rule was omitting the second scenario that you illustrated. 

I have fixed that problem and added a test for that scenario: 
2ff3a5f